spaghetto/src/Spago/Config.purs

```diff
@@ -87,35 +89,80 @@ type PackageConfig =
   , publish :: Maybe PublishConfig
   }
 
+packageConfigCodec :: JsonCodec PackageConfig
+packageConfigCodec = CAR.object "PackageConfig"
```

Comment:
You don't need this to be in any explicit order, do you? I'd just keep in mind that the record codecs (as with the argonaut ones) are alphabetically ordered by field name, so if you're expecting a specific field order you'll need to encode that manually (see `Registry.Manifest`).

Reply:
Ah, good point. I don't think I personally care to order fields, do you think the config should be sorted?

Reply:
If this is something the user is going to write then ideally we’d preserve their ordering in the source — if not, then no I don’t think it matters

Reply:
Ah, I'm afraid that for this codecs that are dealing with Yaml we'd be in the hands of the underlying Yaml library, so it would not matter the order that we set here. I'd need to double check.

Are we able to specify a codec that preserves users' insertion order?

Reply:
We could write one that encodes an `Array (Tuple String Json)` into an object ordered the same as an array, so if you store the user's field order somewhere this codec would preserve that order.

Reply:
Or we could always sort it, for the time being, in an ordering that would be typical for a config.

Reply:
It looks like the yaml library that we use does not sort things by default, [and that can be configured](https://eemeli.org/yaml/#schema-options). I had my configs sorted though, probably by the editor. I am afraid that the same thing would happen to users, rendering our nice sorting useless and even counterproductive. So let's just dump things as is for now, and look into this sorting at a later stage.